nextroute/schema/fleet.go

```diff
@@ -213,48 +245,210 @@ func (fleetInput FleetInput) ToNextRoute() (Input, error) {
 			InitialStops:            &newBacklog,
 			ActivationPenalty:       &v.InitializationCost,
 			ID:                      v.ID,
-			StartLevel:              nil,
+			StartLevel:              startLevel,
 			CustomData:              nil,
 			MinStops:                nil,
 			MinStopsPenalty:         nil,
 			MaxWait:                 nil,
 		}
 	}
 
-	for i, s := range fleetInput.Stops {
-		if _, ok := backlogStops[s.ID]; !ok && s.CompatibilityAttributes == nil {
-			fleetInput.Stops[i].CompatibilityAttributes = &stopCompats
-		}
-	}
+	// for i, s := range fleetInput.Stops {
+	// 	if _, ok := backlogStops[s.ID]; !ok && s.CompatibilityAttributes == nil {
+	// 		fleetInput.Stops[i].CompatibilityAttributes = &stopCompats
+	// 	}
+	// }
 
 	// Create stops with legacy backlog feature.
 	stops := createStops(fleetInput, backlogStops)
 
 	// Put new input format together and return it.
-	input.StopGroups = &fleetInput.StopGroups
-	input.DurationGroups = &fleetInput.DurationGroups
+	if fleetInput.StopGroups != nil {
+		input.StopGroups = &fleetInput.StopGroups
+	}
+	if fleetInput.DurationGroups != nil {
+		input.DurationGroups = &fleetInput.DurationGroups
+	}
 	input.Vehicles = vehicles
 	input.Stops = stops
 
 	return input, nil
 }
 
+// func stopDefaults(fleetInput FleetInput, input Input, stopCompats []string) (Input, []string) {
```

Comment:
dead code

Reply:
sorry, was not expecting you to look so quickly :D fixed just now with another push.